core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -655,6 +821,15 @@ func (s *SmartContract) IsFungibleAssetLocked(ctx contractapi.TransactionContext
 		return false, err
 	}
 
+	txCreator, err := ctx.GetStub().GetCreator()
+	if err != nil {
+		return false, fmt.Errorf("unable to get the transaction creator information: %+v", err)
+	}
+
+	if assetLockVal.Locker != string(txCreator) && assetLockVal.Recipient != string(txCreator) {
```

Comment:
I don't think we need this check. I'd originally added it in the base class, but after a discussion with you and Ermyas, remember removing this logic. I think it's OK to return lock info even if the caller is not a party to the agreement.

Reply:
Now removed this check in all the query functions @VRamakrishna .